homeassistant/components/freebox/device_tracker.py

```diff
@@ -1,65 +1,174 @@
-        # Test the router is accessible.
-        data = await self.connection.lan.get_hosts_list()
-        self.success_init = data is not None
-
-    async def async_scan_devices(self):
-        """Scan for new devices and return a list with found device IDs."""
-        await self.async_update_info()
-        return [device.id for device in self.last_results]
-
-    async def get_device_name(self, device):
-        """Return the name of the given device or None if we don't know."""
-        name = next(
-            (result.name for result in self.last_results if result.id == device), None
+    """Old way of setting up the platform."""
+    pass
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the device_tracker."""
+    fbx = hass.data[DOMAIN]
+
+    entities = []
+
+    for device in fbx.devices.values():
+        entities.append(device)
+
+    async_add_entities(entities)
+
+
+class FreeboxDevice(TrackerEntity):
+    """Representation of a Freebox device."""
+
+    def __init__(self, device: Dict[str, any]):
+        """Initialize a Freebox device."""
+        self._name = device["primary_name"].strip() or DEFAULT_DEVICE_NAME
+        self._mac = device["l2ident"]["id"]
+        self._manufacturer = device["vendor_name"]
+        self._icon = icon_for_freebox_device(device)
+        self._unsub_dispatcher = None
+
+        self.update_state(device)
+
+    def update_state(self, device: Dict[str, any]) -> None:
+        """Update the Freebox device."""
+        self._active = device["active"]
+        if device.get("attrs") is None:
+            # device
+            self._reachable = device["reachable"]
+            self._attrs = {
+                "reachable": self._reachable,
+                "last_time_reachable": datetime.fromtimestamp(
+                    device["last_time_reachable"]
+                ),
+                "last_time_activity": datetime.fromtimestamp(device["last_activity"]),
+            }
+        else:
+            # router
+            self._attrs = device["attrs"]
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self.mac
+
+    @property
+    def name(self) -> str:
+        """Return the name."""
+        return self._name
+
+    @property
+    def latitude(self):
+        """Return the latitude."""
+        if self.active:
+            return self.hass.config.latitude
+        return None
+
+    @property
+    def longitude(self):
+        """Return the longitude."""
+        if self.active:
+            return self.hass.config.longitude
+        return None
+
+    @property
+    def source_type(self) -> str:
+        """Return the source type."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def mac(self) -> str:
+        """Return the MAC address."""
+        return self._mac
+
+    @property
+    def manufacturer(self) -> str:
```

Comment:
Please remove all non abstract properties and just use internal instance attributes for these things instead.